src/System.IO.MemoryMappedFiles/tests/MemoryMappedFile.CreateFromFile.Tests.cs

```diff
@@ -0,0 +1,810 @@
+            using (TempFile file = new TempFile(GetTestFilePath()))
+            {
+                Assert.Throws<ArgumentException>(() => MemoryMappedFile.CreateFromFile(file.Path, FileMode.Truncate));
+            }
+        }
+
+        /// <summary>
+        /// Tests invalid arguments to the CreateFromFile access parameter.
+        /// </summary>
+        [Fact]
+        public void InvalidArguments_Access()
+        {
+            // Out of range access values with a path
+            Assert.Throws<ArgumentOutOfRangeException>("access", () => MemoryMappedFile.CreateFromFile(GetTestFilePath(), FileMode.Open, CreateUniqueMapName(), 4096, (MemoryMappedFileAccess)(-2)));
+            Assert.Throws<ArgumentOutOfRangeException>("access", () => MemoryMappedFile.CreateFromFile(GetTestFilePath(), FileMode.Open, CreateUniqueMapName(), 4096, (MemoryMappedFileAccess)(42)));
+
+            //  Write-only access is not allowed on maps (only on views)
+            Assert.Throws<ArgumentException>("access", () => MemoryMappedFile.CreateFromFile(GetTestFilePath(), FileMode.Open, CreateUniqueMapName(), 4096, MemoryMappedFileAccess.Write));
+
+            // Test the same things, but with a FileStream instead of a path
+            using (TempFile file = new TempFile(GetTestFilePath()))
+            using (FileStream fs = File.Open(file.Path, FileMode.Open))
+            {
+                // Out of range values with a stream
+                Assert.Throws<ArgumentOutOfRangeException>("access", () => MemoryMappedFile.CreateFromFile(fs, CreateUniqueMapName(), 4096, (MemoryMappedFileAccess)(-2), HandleInheritability.None, true));
+                Assert.Throws<ArgumentOutOfRangeException>("access", () => MemoryMappedFile.CreateFromFile(fs, CreateUniqueMapName(), 4096, (MemoryMappedFileAccess)(42), HandleInheritability.None, true));
+
+                // Write-only access is not allowed
+                Assert.Throws<ArgumentException>("access", () => MemoryMappedFile.CreateFromFile(fs, CreateUniqueMapName(), 4096, MemoryMappedFileAccess.Write, HandleInheritability.None, true));
+            }
+        }
+
+        /// <summary>
+        /// Tests various values of FileAccess used to construct a FileStream and MemoryMappedFileAccess used
+        /// to construct a map over that stream.  The combinations should all be valid.
+        /// </summary>
+        [Theory]
+        [InlineData(FileAccess.ReadWrite, MemoryMappedFileAccess.Read)]
+        [InlineData(FileAccess.ReadWrite, MemoryMappedFileAccess.ReadWrite)]
+        [InlineData(FileAccess.ReadWrite, MemoryMappedFileAccess.CopyOnWrite)]
+        [InlineData(FileAccess.Read, MemoryMappedFileAccess.Read)]
+        [InlineData(FileAccess.Read, MemoryMappedFileAccess.CopyOnWrite)]
+        public void FileAccessAndMapAccessCombinations_Valid(FileAccess fileAccess, MemoryMappedFileAccess mmfAccess)
+        {
+            const int Capacity = 4096;
```

Comment:
You've got Capacity separately declared in multiple locations as 4096. Might be better to just move it up with a name like DefaultCapacity so it's only declared in one place.